kernel/src/device/mod.rs

```diff
@@ -41,15 +40,10 @@ pub fn init() -> Result<()> {
     add_node(console, "console")?;
     let tty = Arc::new(tty::TtyDevice);
     add_node(tty, "tty")?;
-    cfg_if! {
-        if #[cfg(all(target_arch = "x86_64", feature = "cvm_guest"))] {
-            let tdx_guest = Arc::new(tdxguest::TdxGuest);
-
-            if tdx_is_enabled() {
-                add_node(tdx_guest, "tdx_guest")?;
-            }
-        }
-    }
+    #[cfg(target_arch = "x86_64")]
+    if_tdx_enabled!({
+        add_node(Arc::new(tdxguest::TdxGuest), "tdx_guest")?;
+    });
```

Comment:
It seems that the check of `target_arch = "x86_64"` should be built into the macro.